lib/wor/paginate/utils/preserve_records_helper.rb

```diff
@@ -0,0 +1,49 @@
+require_relative 'uri_helper'
+require_relative 'preserve_modes'
+
+module Wor
+  module Paginate
+    module Utils
+      class PreserveRecordsHelper
+        def initialize(content, url, options)
+          @content = content
+          @url = url
+          @options = options
+        end
+
+        def call
+          [content.where("#{field} <= :last_value", last_value: last_value),
+           UriHelper.replace_query_params(url, query_param_name => last_value)]
+        end
+
+        private
+
+        attr_reader :content, :url, :options
+
+        def by
+          @by ||= begin
+            by = options[:by]&.to_s || 'timestamp'
+            raise ArgumentError, "'by' option shuld be 'id' or 'timestamp'" unless
```

Comment:
could you change this typo to "should" please?

Reply:
```suggestion
            raise ArgumentError, "'by' option should be 'id' or 'timestamp'" unless
```